pil_analyzer/src/pil_analyzer.rs

```diff
@@ -26,15 +27,23 @@ pub fn process_pil_file<T: FieldElement>(path: &Path) -> Analyzed<T> {
     analyzer.condense()
 }
 
+pub fn process_pil_ast<T: FieldElement>(pil_file: PILFile<T>) -> Analyzed<T> {
+    let mut analyzer = PILAnalyzer::new();
+    analyzer.process_ast(pil_file);
+    analyzer.condense()
+}
+
 pub fn process_pil_file_contents<T: FieldElement>(contents: &str) -> Analyzed<T> {
     let file_name = Path::new("input");
     let mut analyzer = PILAnalyzer::new();
 
-    analyzer.process(
-        [(file_name.to_owned(), ParsedFile::parse(file_name, contents))]
-            .into_iter()
-            .collect(),
-    );
+    let ast = parser::parse(Some("input"), contents).unwrap_or_else(|err| {
```

Comment:
I'm wondering - maybe "stdin" or "unspecified input" could be a separate enum value for file name? But we could also do this another time.

Reply:
I was using `Option<SourceRef>` at first, so that `None` would represent "generated" statements... but I got the feeling I was overcomplicating for the current usage (was having to decide when to propagate the Option or unwrap).
Using an enum could be nice for a future PR.